src/Uno.Extensions.Navigation.Generators/ForceBindingsUpdateGenerationContext.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis;
+using Uno.Extensions.Generators;
+using Uno.Extensions.Navigation.UI.Controls;
+
+namespace Uno.Extensions.Navigation.Generators;
+
+internal record ForceBindingsUpdateGenerationContext(
+	GeneratorExecutionContext Context,
+
+	// Core types
+	[ContextType("Uno.Extensions.Navigation.IForceBindingsUpdate")] INamedTypeSymbol ForceBindingsUpdateInterface,
+
+	// Attributes
+	[ContextType(typeof(ForceUpdateAttribute))] INamedTypeSymbol UpdateAttribute,
+
+	// General stuff types
+	[ContextType("Microsoft.UI.Xaml.Controls.Page")] INamedTypeSymbol Page)
+{
+	public bool IsGenerationNotDisable(ISymbol symbol)
+		=> IsGenerationEnabled(symbol) ?? true;
+
+	public bool? IsGenerationEnabled(ISymbol symbol)
+		=> symbol.FindAttributeValue<bool>(UpdateAttribute, nameof(ForceUpdateAttribute.IsEnabled), 0) is { isDefined: true } attribute
+			? attribute.value ?? true
+			: null;
+
+	private IImmutableSet<string>? _XBindFiles;
+	private static Regex ClassRegEx = new Regex("x:Class=\"([\\w.]+)\"");
+	private static string xBind = "{x:Bind ";
+
+	private IImmutableSet<string> XBindFiles
+	{
+		get
+		{
+			if (_XBindFiles is null)
+			{
+				var files = ImmutableHashSet.CreateBuilder<string>();
+				foreach (var file in Context.AdditionalFiles)
+				{
+					using (var reader = new StreamReader(file.Path))
+					{
+						string? className = null;
+						var hasXBind = false;
+						while (
+							!reader.EndOfStream &&
+							(className is null ||
+							!hasXBind))
+						{
+							var txt = reader.ReadLine();
+							if(className is null)
+							{
+								var classNameMatch = ClassRegEx.Match(txt);
+								if (classNameMatch.Success &&
+									classNameMatch.Groups.Count > 1)
+								{
+									className = classNameMatch.Groups[1].Value;
+								}
+							}
+							if (txt is not null &&
+								txt.Contains(xBind))
+							{
+								hasXBind = true;
+							}
+						}
+						if(className is not null && hasXBind)
+						{
+							files.Add(className);
+						}
+					}
+				}
```

Comment:
I'm not sure how this would be more performant that just reading the file. This seems like it would actually take more CPU cycles. The className and hasXBind can probably be moved outside of the ForEach to reduce allocations.

Reply:
It might be worth separating this into own static helper in a separate file to make the code better to deal with.

Reply:
Refactored to extract out static helper method